Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/dccticketing/ui/consent/two/DccTicketingConsentTwoFragmentTest.kt

```diff
@@ -0,0 +1,169 @@
+        setupMockViewModel(
+            object : DccTicketingConsentTwoViewModel.Factory {
+                override fun create(
+                    dccTicketingSharedViewModel: DccTicketingSharedViewModel,
+                    containerId: CertificateContainerId
+                ): DccTicketingConsentTwoViewModel = viewModel
+            }
+        )
+    }
+
+    @Screenshot
+    @Test
+    fun dccTicketingCertificateSelectionFragment_certificates() {
+        every { viewModel.uiState } returns MutableLiveData(
+            DccTicketingConsentTwoViewModel.UiState(
+                dccTicketingTransactionContext = dccTicketingTransactionContext,
+                certificate = mockTestCertificate("Andrea Schneider")
+            )
+        )
+        launchFragmentInContainer2<DccTicketingConsentTwoFragment>(
+            testNavHostController = navController,
+            fragmentArgs = fragmentArgs
+        )
+        takeScreenshot<DccTicketingConsentTwoFragment>()
+    }
+
+    private fun mockTestCertificate(
+        name: String,
+        isPending: Boolean = false,
+        isUpdating: Boolean = false
+    ): TestCertificate = mockk<TestCertificate>().apply {
+        every { headerExpiresAt } returns Instant.now().plus(20)
+        every { isCertificateRetrievalPending } returns isPending
+        every { isUpdatingData } returns isUpdating
+        every { fullName } returns name
+        every { registeredAt } returns Instant.parse("2021-05-21T11:35:00.000Z")
+        every { personIdentifier } returns CertificatePersonIdentifier(
+            firstNameStandardized = "firstNameStandardized",
+            lastNameStandardized = "lastNameStandardized",
+            dateOfBirthFormatted = "1943-04-18"
+        )
+        every { qrCodeToDisplay } returns CoilQrCode(ScreenshotCertificateTestData.testCertificate)
+        every { isValid } returns true
+        every { sampleCollectedAt } returns Instant.parse("2021-05-21T11:35:00.000Z")
+        every { getState() } returns CwaCovidCertificate.State.Valid(headerExpiresAt)
+        every { isNew } returns false
+        every { containerId } returns TestCertificateContainerId("testCertificateContainerId")
+        every { rawCertificate } returns mockk<TestDccV1>().apply {
+            every { test } returns mockk<DccV1.TestCertificateData>().apply {
+                every { testType } returns "LP6464-4"
+            }
+        }
+        every { isPCRTestCertificate } returns false
+    }
+
+    private fun generateDccTicketingQrCodeData(): DccTicketingQrCodeData {
+        return DccTicketingQrCodeData(
+            protocol = "protocol",
+            protocolVersion = "protocol_version",
+            serviceIdentity = "service_identity",
+            privacyUrl = "http://very_privacy_url.de",
+            token = UUID.randomUUID().toString(),
+            consent = "Yes, please",
+            subject = UUID.randomUUID().toString(),
+            serviceProvider = "Service Provider"
```

Comment:
Please use the text from figma? to have the same exact one at least in German screenshots